ipaddr/detect.go

```diff
@@ -0,0 +1,138 @@
+package ipaddr
+
+import (
+	"fmt"
+	"net"
+)
+
+// GetPrivateIPv4 returns the list of private network IPv4 addresses on
+// all active interfaces.
+func GetPrivateIPv4() ([]*net.IPAddr, error) {
```

Comment:
We could probably remove these helpers and change the default config to go-sockaddr templates that accomplishes this as well.

Reply:
That makes mocking the address resolution more difficult. Let me think about a solution.